katdal/__init__.py

```diff
@@ -329,7 +329,14 @@ def open(filename, ref_ant='', time_offset=0.0, **kwargs):
     filenames = [filename] if isinstance(filename, basestring) else filename
     datasets = []
     for f in filenames:
-        dataset = _file_action('__call__', f, ref_ant, time_offset, **kwargs)
+        # V4 RDB file with optional URL-style query string
+        if f.split('?')[0].endswith('.rdb'):
```

Comment:
This does look pretty hacky. For one thing, if someone chooses to put a `?` in their filename, it's going to mysteriously fail. Maybe rather check for `://` or even `.startswith('file:///')`?

Reply:
Of course it's hacky. I called it with `'dump.20180215-0017.rdb?capture_block_id=00000000-00000-1518711320'` and I don't expect anyone else to call it with a ?. This will all be cleared up soon (hopefully after March :-)).

Reply:
How about using
```python
if urlparse.urlsplit(f).path.endswith('.rdb'):
```
At least then it will handle the vagaries of URL parsing. It looks like urlparse is happy to be given a relative URL aka a filename.

Reply:
Done. Although it still throws out a filename with a `?`.